### PR TITLE
Add support for in-band syncing

### PR DESCRIPTION
**Pull request by jpwilliams**
## Summary

Adds in-band sync support by returning different data based on whether a `PUT` request is signed (and validated) or not.

In draft while functionality is built out elsewhere.

## Checklist

- [ ] ~Added a [docs PR](https://github.com/inngest/website) that references this PR~ N/A Accounted for elsewhere
- [x] Added unit/integration tests
- [x] Added changesets if applicable

## Related

- INN-3306

### CODE REVIEW

File: packages/inngest/src/components/InngestCommHandler.ts [hunk 1]
```diff
@@ -1451,6 +1441,134 @@ export class InngestCommHandler<
...
+    }
+
+    const body: InBandRegisterRequest = {
+      app_id: this.client.id,
+      capabilities: registerBody.capabilities,
+      env: introspectionBody.env,
+      framework: registerBody.framework,
+      functions: registerBody.functions,
+      inspection: introspectionBody,
+      platform: "?",
```
Comment (goodoldneon) [inline packages/inngest/src/components/InngestCommHandler.ts:1480]: Is it possible to populate this?

Comment (jpwilliams) [reply packages/inngest/src/components/InngestCommHandler.ts:1480]: Baha yes, this is prior to our discussion. Will change.

File: packages/inngest/src/components/InngestCommHandler.ts [hunk 2]
```diff
@@ -1199,14 +1136,75 @@ export class InngestCommHandler<
...
+            .then((kind) => {
+              return kind === syncKind.InBand;
+            }),
+        ]);
+
+        if (inBandSyncRequested) {
+          // Validation can be successful if we're in dev mode and did not
+          // actually validate a key. In this case, also check that we did indeed
+          // use a particular key to validate.
+          const sigCheck = await signatureValidation;
```
Comment (jpwilliams) [inline packages/inngest/src/components/InngestCommHandler.ts:1171]: This is always true if the SDK's mode is `"dev"`, but will fail if it's `"cloud"` and the request could not be validated.

File: packages/inngest/src/components/InngestCommHandler.ts [hunk 3]
```diff
@@ -1199,14 +1136,75 @@ export class InngestCommHandler<
...
+
+          if (!sigCheck.success) {
+            return {
+              status: 401,
```
Comment (jpwilliams) [inline packages/inngest/src/components/InngestCommHandler.ts:1175]: A failed validation when we are allowing an in-band sync and one has been requested is always a `401`.

File: packages/inngest/src/components/InngestCommHandler.ts [hunk 4]
```diff
@@ -1199,14 +1136,75 @@ export class InngestCommHandler<
...
+              body: stringify({
+                code: "sig_verification_failed",
+                message: "",
+                name: "SigVerificationFailedError",
```
Comment (jpwilliams) [inline packages/inngest/src/components/InngestCommHandler.ts:1179]: @amh4r I used the same JSON as the Python SDK; not sure if this code is used anywhere else yet.

Comment (amh4r) [reply packages/inngest/src/components/InngestCommHandler.ts:1179]: I don't think it's used anywhere else. We should be consistent with the error code across SDKs but the name and message can be excluded

**Review by amh4r — APPROVED** [review]

**Pull request merged**